src/nodes.rs

```diff
@@ -135,17 +135,87 @@ impl<'a, T> IntoIterator for &'a NodeMap<T> {
     }
 }
 
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct BoolNodeMap(bit_vec::BitVec<u32>);
+
+impl BoolNodeMap {
+    pub fn new_with_len(len: NodeCount) -> Self {
+        BoolNodeMap(bit_vec::BitVec::from_elem(len.0, false))
+    }
+
+    pub fn set(&mut self, i: NodeIndex, value: bool) {
+        self.0.set(i.0, value);
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    pub fn iter(&self) -> impl Iterator<Item = bool> + '_ {
+        self.0.iter()
+    }
+
+    pub fn true_indices(&self) -> impl Iterator<Item = NodeIndex> + '_ {
+        self.iter()
+            .enumerate()
+            .filter_map(|(i, b)| if b { Some(i.into()) } else { None })
+    }
+}
+
+impl Encode for BoolNodeMap {
+    fn encode_to<T: Output + ?Sized>(&self, dest: &mut T) {
+        (self.0.len() as u32).encode_to(dest);
+        self.0.to_bytes().encode_to(dest);
+    }
+}
+
+impl Decode for BoolNodeMap {
+    fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
+        let len = u32::decode(input)? as usize;
+        let bytes = Vec::decode(input)?;
+        let mut bv = bit_vec::BitVec::from_bytes(&bytes);
+        bv.truncate(len);
+        Ok(BoolNodeMap(bv))
+    }
+}
+
+impl FromIterator<bool> for BoolNodeMap {
+    fn from_iter<T: IntoIterator<Item = bool>>(iter: T) -> Self {
+        BoolNodeMap(bit_vec::BitVec::from_iter(iter))
+    }
+}
+
+impl Index<NodeIndex> for BoolNodeMap {
+    type Output = bool;
+
+    fn index(&self, vidx: NodeIndex) -> &bool {
+        &self.0[vidx.0 as usize]
+    }
+}
+
 #[cfg(test)]
 mod tests {
-    use crate::nodes::NodeIndex;
+    use crate::nodes::{BoolNodeMap, NodeIndex};
     use codec::{Decode, Encode};
     #[test]
-    fn decoding_works() {
+    fn decoding_node_index_works() {
         for i in 0..1000 {
             let node_index = NodeIndex(i);
             let mut encoded: &[u8] = &node_index.encode();
             let decoded = NodeIndex::decode(&mut encoded);
             assert_eq!(node_index, decoded.unwrap());
         }
     }
+
+    #[test]
+    fn decoding_bool_node_map_works() {
+        let bool_node_map = BoolNodeMap([true, false, true, true, true].iter().cloned().collect());
+        let encoded: Vec<_> = bool_node_map.encode();
+        let decoded = BoolNodeMap::decode(&mut encoded.as_slice()).expect("decode should work");
+        assert_eq!(decoded, bool_node_map);
+    }
```

Comment:
Can we have one additional test that checks that the encoding of a BoolNodeMap of length N is close to N/8 rather than N? Just to be sure :P

Reply:
Done :)